environment_switcher/lib/src/environment_switcher.dart

```diff
@@ -0,0 +1,195 @@
+import 'package:flutter/material.dart';
+
+import 'environment.dart';
+import 'environment_store.dart';
+import 'select_environment_sheet.dart';
+
+export 'environment.dart';
+export 'environment_store.dart';
+
+class EnvironmentSwitcher extends StatefulWidget {
+  final Widget Function(Environment) builder;
```

Comment:
`BuildContext` may also be required for a generic solution

Reply:
Why is that? Wouldn't the `BuildContext` from the `build()` method be fine?

Reply:
Because it's using a context from higher up the tree. Take for example:

```
Widget build(BuildContext contextA) {
  return BlocProvider<FooBloc>(
    bloc: _fooBloc,
    child: EnvironmentSwitcher(
      builder: (env) {
        final fooBloc = BlocProvider.of<FooBloc>(contextA);
        return MyWidget(env: env);
      }),
    )
  );
}
```

Which would throw because `contextA` doesn't contain a `BlocProvider`.

It's not required where it's used in SP, but could lead to unexpected errors in future apps. This is why builder functions typically pass back a context.

Reply:
Ah yes! that makes sense. I misunderstood where you were referring to put the `BuildContext`. I'll do that now.